server/container_create_linux.go

```diff
@@ -556,19 +556,31 @@ func (s *Server) mountImage(ctx context.Context, specgen *generate.Generator, im
 		log.Debugf(ctx, "Using final mount path: %s", mountPoint)
 	}
 
+	upperDirPath := filepath.Join(imageVolumesPath.work, containerID, imageID, "upper")
+	workDirPath := filepath.Join(imageVolumesPath.work, containerID, imageID, "work")
+
+	for _, dir := range []string{upperDirPath, workDirPath} {
+		if err := os.MkdirAll(dir, 0o700); err != nil {
+			return nil, nil, fmt.Errorf("create image volume path: %w", err)
+		}
+	}
+
+	mountOpt := fmt.Sprintf(
+		"lowerdir=%s:%s,upperdir=%s,workdir=%s",
+		mountPoint, imageVolumesPath.mounts, upperDirPath, workDirPath,
+	)
```

Comment:
@giuseppe PTAL from a security standpoint.

Reply:
thanks for the tag, I don't see anything wrong in this from a security PoV

Reply:
Thank you! :pray: